### ISSUE TITLE
Preview scroll height is out of sync with editor when images are embedded

### ISSUE THREAD

**Issue by cttttt** [careteditor/caret#105]
# What do I do?  What happens?

- Create a new markdown document in Caret **1.5.4** and add a few screen heights worth of text to the buffer.
- Use Command-R to open the preview to the right of the buffer.
- NOTICE: While typing, the preview on the right scrolls with the text on the left.
- Save an image that's taller than a few lines of text to disk.
- Key in the markdown to embed the image into the buffer (i.e. key in `![caption](path-to-image.png)` on its own line).
- NOTICE: The image appears at the bottom of the preview on the right.
- Scroll to the top of the text buffer.
- NOTICE: The preview scrolls all the way to the top of the document.
- Scroll to the bottom of the buffer and type in a paragraph or two of text below the image.
- **ISSUE: The preview scrolls down only far enough to reveal half of the inserted image, but not far enough to reveal the text being keyed into the buffer.  Even if I scroll the buffer all the way down to the bottom, that text below the image is not visible.**
- **ISSUE: Also, embedding more images into the document causes more of a disconnect between the scroll position of the buffer on the left and the preview on the right**

# What did I expect?

- For the preview to scroll to reveal the text I'm currently keying in.  The height of inserted images should be considered when setting the scroll position of the preview pane.
- Also, I'd expect the scroll position to reveal the text *at the insertion point* of the buffer while entering text...but, at the same time, for the preview to scroll as the buffer is scrolled.  I'm probably asking for something that isn't even practical.  Please keep me honest :)

# Workarounds

- A partial workaround is to just scroll the preview on the right down to the bottom.  It's kinda annoying, though, because doing things in the buffer on the left causes the preview to scroll back to a position that hides the text below the last image.

# What could help?

- An improved algorithm for determining the scroll height of the preview.  It feels like the current algorithm assumes the height of all images is a single line, but it ought to be derived from the image...somehow.
- An option to turn off the scrollbinding between the preview and the buffer.
- An option to scroll both the buffer and preview such that the current section is up top.

Please let me know if you need any more info,

~ chris

**Comment by erusev** [comment]
Thanks, Chris, that's very well written, we'll see what we can do.

**Comment by astoilkov** [comment]
Hi @cttttt,

We resolved this issue. The fix is available in the latest 1.6.1 Caret release. Could you try it out and post here the results so we can close the issue.

**Comment by astoilkov** [comment]
Closing this. Do not hesitate to reopen the issue if there is problems with this scenario.

**Comment by cttttt** [comment]
@astoilkov , just confirmed that when it comes to images, scrolling is on point.  Thanks a lot!

**Comment by astoilkov** [comment]
What about the first issue you have logged?

**Comment by cttttt** [comment]
@astoilkov just double checked.

- The first *ISSUE* in this particular bug report seems resolved.  If I insert an image at the bottom of the document, the preview scrolls to entirely reveal the image and the empty line beneath it.  Continuing to type shows the inserted text and preview continues scrolling down as I insert newlines and such.
- The second *ISSUE* above, the disconnect between scroll height in the preview and the scroll position of the buffer also seems fixed.  As a workaround, in the documents I've been editing, I've inserted links to image instead of the actual inline images.  I just went in and replaced those with embedded images and the scroll height seems to line up between the preview and the buffer.
- As for the other bug report, ...I dropped a comment there, issue #106 .  It's resolved.

Thanks again!

~ chris
